preproc.py

```diff
@@ -44,3 +44,15 @@ def compute_dc_meta(vg, vd, ids):
     scale = {'vg':vg_scale, 'vd':vd_scale, 'id':id_scale}
 
     return scale, vg_shift
+
+def truncate(data_arrays, truncate_range, axis):
+    # The data within the truncate_range will be removed, the rest will be returned.
```

Comment:
Just two quick fix:
1. add an assert if axis is out of range
2. sorry I forget to tell you truncate_range should be exclusive on the lower bound and inclusive on the upper bound, means e > lower and e <= upper